Mhv 33532 landing page sort

alex-morgun-1 opened the pull request:
## Description
On messages landing page (auth view) /my-health/secure-messages ,adding an ability to sort messages by sent date (ascending + descending), and by senders name in alphabetical order (A to Z+ Z to A)

## Original issue(s)
[JIRA - SM to VA.gov:  Message landing page (auth)-Sort and/or filter development](https://vajira.max.gov/browse/MHV-33532)


## Testing done


## Screenshots
![image](https://user-images.githubusercontent.com/87077843/187769455-f9b9f29e-62e4-42f0-b733-37abd32c05a8.png)


## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

micahchiang committed:
initial redux code

micahchiang committed:
merge master

micahchiang committed:
inbox view and message card

micahchiang committed:
pagination, sorting

mmoyer-va committed:
Merge branch 'main' into mhv-sm-main

mmoyer-va committed:
MHV-33575 Changed import to fix error 'node_modules\web-components doesn't exist'

mmoyer-va committed:
Merge remote-tracking branch 'origin/main' into mhv-sm-main

mmoyer-va committed:
[#MHV-33575] Temporarily removing MHV SM authentication to facilitate development

mmoyer-va committed:
[#MHV-33798] Added some routes for SM app

Commit:
Successfully wrote and ran passing simple test that renders header without errors

Commit:
erased comment

Commit:
Merge branch 'MHV-33578-first-unit-test' into mhv-sm-main

KolbyVA committed:
[#MHV-33798] First work, just got write access

KolbyVA committed:
[MHV-33798] First version of Reply Form

Commit:
Created menu from desktop view on the va.gov design page and edited to display messages list, still need to add font awesome icon and need to switch the close (x) button on next to myHealth as opposed to next to Messages

mattwrightva committed:
MHV-33554: Started MHV SM compose new message page

mattwrightva committed:
MHV-33554:  Basic mobile compose structure finished

KolbyVA committed:
[MHV-33798] Mobile Reply view done, some components need work, but are shared with other containers

mattwrightva committed:
Merge branch 'MHV-33554-mobile-compose-structure' into mhv-sm-main

KolbyVA committed:
Merge branch 'MHV-33798-reply-page-mobile' into mhv-sm-main

Commit:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into MHV-33796-Mobile-Menu-Navigation

Commit:
Finished styling the menu, just need it to not close when mobile breakpoint is hit

mattwrightva committed:
MHV-33584: Compose structure / layout for desktop complete

mattwrightva committed:
MHV-33584: Tweaks to compose desktop structure/layout

mmoyer-va committed:
Merge branch 'main' into mhv-sm-main

KolbyVA committed:
Desktop view done

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33584-compose-structure-desktop

KolbyVA committed:
[#MHV-33802] Breadcrumbs layout + duplicate error

KolbyVA committed:
Merge branch 'mhv-sm-main' into MHV-33799-reply-page-desktop

mattwrightva committed:
MHV-33584: Unit tests added for Compose container and ComposeForm component

alex-morgun-1 committed:
[#MHV-33546] message detail page - mobile

alex-morgun-1 committed:
Merge branch 'MHV-33546-message-detail-mobile' into mhv-sm-main

Commit:
Menu now docks to the top right, removed icon component and instead used vas class name of va-btn-close-icon which adds the icon instead

mattwrightva committed:
MHV-33584: Unit tests fixed for ComposeForm and written for MessageCategoryInput component

Commit:
Removed navigation from routes

mattwrightva committed:
MHV-33584: Tweaks to compose views and created compose scss file

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33584-compose-structure-desktop

mmoyer-va committed:
[#MHV-33533] Added conditionally displayed auth/unauth landing pages

mmoyer-va committed:
Merge branch 'mhv-sm-main' into MHV-33533-landing-page-layout

Commit:
Issue with unit test not passing was scss import, ended up putting all of my scss into the main css file and removing the scss import, now the unit test passes

Commit:
changed the unit test to check for nav bar header

Commit:
seperated navigation styling from from secure-messageing.scss and imported it instead as a seperate sass syle sheet

mattwrightva committed:
MHV-33594: Search messages mobile view complete plus other fixes

mattwrightva committed:
MHV-33594: Deleted lowercase breadcrumbs component

mattwrightva committed:
MHV-33594: Search messages mobile and desktop view complete

mattwrightva committed:
MHV-33801: Search results mobile and desktop layout complete

Commit:
Cleaned up ordering of elements, class naming and styling

Commit:
cleaned up the style sheet

Commit:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33796-mobile-menu-navigation

Commit:
removed style that hid the x button

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33801-search-results-mobile-view

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33594-search-messages-mobile-view-structure

alex-morgun-1 committed:
[#MHV-33785] message details desktop

alex-morgun-1 committed:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33785-message-detail-desktop

alex-morgun-1 committed:
Merge branch 'mhv-sm-main' of https://github.com/department-of-veterans-affairs/vets-website into mhv-33785-message-detail-desktop

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33594-search-messages-mobile-view-structure

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33801-search-results-mobile-view

mattwrightva committed:
Merge branch 'mhv-sm-main' into MHV-33594-search-messages-mobile-view-structure

mattwrightva committed:
MHV-33801: Renamed breadcrumbs to pascal case

mmoyer-va committed:
[#MHV-33533] Built out the MHV SM landing page

mmoyer-va committed:
Merge remote-tracking branch 'origin' into MHV-33533-landing-page-layout

mmoyer-va committed:
Merge branch 'mhv-sm-main' into MHV-33533-landing-page-layout

# Conflicts:
#	src/applications/mhv/secure-messaging/components/breadcrumbs.jsx

mmoyer-va committed:
[#MHV-33533] Fixed breadcrumbs import

mmoyer-va committed:
[#MHV-33533] Fixed styling on the message-sort control

mmoyer-va committed:
[#MHV-33533] Organized components

mmoyer-va committed:
[#MHV-33533] Updated inbox styling to better match designs

KolbyVA committed:
[#MHV-35261] added unit tests for reply page

KolbyVA committed:
Merge branch 'mhv-sm-main' into MHV-35261-reply-page-unit-test

mmoyer-va committed:
Merge branch 'mhv-sm-main' into MHV-33533-landing-page-layout

alex-morgun-1 committed:
fix message details note typo

alex-morgun-1 committed:
[#MHV-33532] message sorting feature on InboxListView component

sorting messages on a component level, given component accepts an array of message data as a prop. Additionally, modifying mockdata sent dates temporarily until api calls are established

alex-morgun-1 committed:
Merge branch 'main' into mhv-33532-landing-page-sort

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/components/Navigation.jsx:
```diff
@@ -1,92 +1,29 @@
+    <ul>
+      <li>
+        <div className="nav-title">
+          <i className="medkit-icon fas fa-medkit" />
```
Since our team has been working in a separate dedicated branch, we are now submitting individual pull requests with outstanding work from all our team members. This component is modified and will be updated in one of the following PRs shortly. Please disregard this

alex-morgun-1 committed:
removing unused unit test

Header component is no longer used. Somehow in between the merges, this unit test got left behind on this branch, but is already killed on the main. But this unit test fails a PR check since the component is missing
